tests/specs/bullrun.spec.js

```diff
@@ -0,0 +1,237 @@
+import { applicationProxy, getMockDeviceEvent, restoreUserData } from "../applicationProxy";
+import OnboardingPage from "../po/onboarding.page";
+import ModalPage from "../po/modal.page";
+import GenuinePage from "../po/genuine.page";
+import PasswordPage from "../po/password.page";
+import AnalyticsPage from "../po/analytics.page";
+import PortfolioPage from "../po/portfolio.page";
+import data from "../data/onboarding/";
+import {
+  deviceInfo155 as deviceInfo,
+  mockListAppsResult,
+} from "@ledgerhq/live-common/lib/apps/mock";
+
+jest.setTimeout(600000);
+
+describe("When I launch the app for the first time", () => {
+  let app;
+  let onboardingPage;
+  let modalPage;
+  let genuinePage;
+  let passwordPage;
+  let analyticsPage;
+  let portfolioPage;
+  let mockDeviceEvent;
+
+  beforeAll(async () => {
+    app = await applicationProxy({ MOCK: true, DISABLE_MOCK_POINTER_EVENTS: true });
+    onboardingPage = new OnboardingPage(app);
+    modalPage = new ModalPage(app);
+    genuinePage = new GenuinePage(app);
+    passwordPage = new PasswordPage(app);
+    analyticsPage = new AnalyticsPage(app);
+    portfolioPage = new PortfolioPage(app);
+    mockDeviceEvent = getMockDeviceEvent(app);
+
+    return app.start();
+  });
+
+  afterAll(() => {
+    return restoreUserData();
```

Comment:
a better duality would be to put the backupUserData() in beforeAll no?

On the pattern involved here itself. I think it would be way better to spawn Spectron with a specific user data for the run of this test.

to my mind, each single test should run in a standalone user data that gets created somewhere like in a /tmp/folder , that way it's not touching existing user data at all.

Reply:
The rimraf was what was present before I just added an imperfect backup, I'll look into a better way of doing this.